gaurav/src/redux/reducers.ts

```diff
@@ -0,0 +1,33 @@
+interface PunchData {
+  time: string;
+}
+interface IpunchState {
+  punchData: PunchData[]
+  punchInDate: string | null;
+  punchOutDate: string | null;
+}
+const initialState: IpunchState = {
+  punchData: [],
+  punchInDate: null,
+  punchOutDate: null,
+};
+const dataReducer = (state: IpunchState = initialState, action: any): IpunchState => {
+  switch (action.type) {
+    case "PUNCH_IN":
+      return {
+        ...state,
+        punchInDate: action.payload,
+        punchData: [...state.punchData, { time: action.payload }]
+      };
+    case "PUNCH_OUT":
+      return {
+        ...state,
+        punchOutDate: action.payload,
+        punchData: [...state.punchData, { time: action.payload }]
```

Comment:
Can you add an example data for the contents of `punchData`

Reply:
when user `punchIn`  or `punchOut` then that time will be added to the `punchData []` array.
and then we `map` the `punchData[]` to show the `punchIn` and `punchOut` timew based on the `index.`

![Screenshot from 2024-11-26 23-00-12](https://github.com/user-attachments/assets/192b0bea-c2d8-4245-a47d-83f6f57fbfbf)

Reply:
I meant the data in JSON format @gauravThakur-sgl

Reply:
punchData in JSON format : 

![Screenshot from 2024-11-27 16-07-20](https://github.com/user-attachments/assets/5517525f-f418-41af-8780-8f58f608b448)